### PR TITLE
customize strings

### PR DESCRIPTION
**Pull request by ABaldwinHunter**
We need to nest the `strings` config option under a wildcard engine `config` key that `cc/yaml` allows.

@codeclimate/review  @pbrisbin

### CODE REVIEW

File: lib/fix-me.js [hunk 1]
```diff
@@ -18,10 +18,10 @@ FixMe.prototype.runEngine = function(){
...
 
   if (fs.existsSync('/config.json')) {
-    var overrides = JSON.parse(fs.readFileSync('/config.json'));
-
-    for (var prop in overrides) {
-        config[prop] = overrides[prop];
+    var user_config = JSON.parse(fs.readFileSync('/config.json'));
+    config.include_paths = user_config['include_paths'];
+    if (user_config.config && user_config['config']['strings']) {
+      config.strings = user_config['config']['strings']
```
Comment (pbrisbin) [inline lib/fix-me.js]: I think we can keep the `for (var ..` form in case we want to add new config keys later.

WDYT about this:

```js
config.include_paths = user_config.include_paths

if (user_config.config) {
  for (var prop in user_config.config) {
    config[prop] = user_config.config[prop]
  }
}
```

**Comment by ABaldwinHunter** [comment]
I plan to update the `.codeclimate.yml` settings for this engine in a soon to follow PR.

File: lib/fix-me.js (hunk 1 above)
Comment (GordonDiggs) [inline lib/fix-me.js]: :lipstick: missing semicolon

Comment (ABaldwinHunter) [inline lib/fix-me.js]: Okay that sounds good

File: lib/fix-me.js [hunk 2]
```diff
@@ -18,10 +18,10 @@ FixMe.prototype.runEngine = function(){
       self = this;
 
   if (fs.existsSync('/config.json')) {
-    var overrides = JSON.parse(fs.readFileSync('/config.json'));
-
-    for (var prop in overrides) {
-        config[prop] = overrides[prop];
+    var user_config = JSON.parse(fs.readFileSync('/config.json'));
```
Comment (wfleming) [inline lib/fix-me.js]: nit: we generally use snakeCasing for var names in JS (except in cases where it's a key that maps to JSON or something).

**Comment by pbrisbin** [comment]
I agree with @wfleming 's nit about `userConfig`. Also, the commit message could be a little more accurate -- this doesn't add that feature, it fixes a bug in it. Other than that, this is looking good.

**Comment by ABaldwinHunter** [comment]
@pbrisbin Good call. Updating message.

@wfleming good catch, I agree about the casing. Just noticed that also, but was planning to address in a separate PR - and fix the `include_paths` sprinkled throughout as well. 

Adding as a separate commit here.

File: lib/fix-me.js [hunk 3]
```diff
@@ -12,20 +12,24 @@ function FixMe() { }
 FixMe.prototype.runEngine = function(){
   var analysisFiles = [],
       config = {
-        include_paths: ["./"],
+        includePaths: ["./"],
```
Comment (pbrisbin) [inline lib/fix-me.js]: I would probably leave all the `include_paths` as-is, personally. Loosely speaking they do map to JSON keys:

> (except in cases where it's a key that maps to JSON or something)

Also, I'd probably characterize that as drive-by, whereas `userConfig` is something you're adding now, so it should follow the guide.

File: lib/fix-me.js [hunk 4]
```diff
@@ -12,20 +12,24 @@ function FixMe() { }
...
       },
       self = this;
 
   if (fs.existsSync('/config.json')) {
-    var overrides = JSON.parse(fs.readFileSync('/config.json'));
+    var userConfig = JSON.parse(fs.readFileSync('/config.json'));
 
-    for (var prop in overrides) {
-        config[prop] = overrides[prop];
+    config.includePaths = userConfig['include_paths'];
```
Comment (pbrisbin) [inline lib/fix-me.js]: I think you should use object notation whenever possible (e.g. `userConfig.include_paths`). It's only when the key is a variable (as in `config[prop]`) that we need to use `[]`-notation.

Comment (ABaldwinHunter) [inline lib/fix-me.js]: okay :+1:

**Comment by ABaldwinHunter** [comment]
@pbrisbin @wfleming Thanks for feedback.

Ready for another look!

**Comment by pbrisbin** [comment]
:clap: LGTM!
